src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -25,33 +27,40 @@ public class ScanResultToComponentListTransformer {
      * @param rapidScanFullResults
      * @return list of {@link Component}s
      */
-    public List<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
+    public Set<Component> transformScanResultToComponentList(List<DeveloperScansScanView> rapidScanFullResults) {
         HashMap<String, ScanMetadata> componentIdWithMetadata = new HashMap<>();
-
+        Set<String> componentNamesWithNullIds = new HashSet<>(rapidScanFullResults.size());
         for (DeveloperScansScanView component : rapidScanFullResults) {
-            componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+            if (component.getExternalId() == null) {
+                componentNamesWithNullIds.add(component.getComponentName());
+            } else {
+                componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+            }
```

Comment:
Handle NULL external ID value scenario to avoid NPE. Also, collect the component names to print them once each.

Reply:
Which of the tickets in the description refer to this NPE?

Reply:
IDETECT-3974 in last comment.

Reply:
I see , thank you